app/app.py

```diff
@@ -1,48 +1,150 @@
+
+    return manager
+
+
+def add_handlers(app):
+    @app.errorhandler(404)
+    def handle404(e):
+        return render_template('errors/404.html'), 404
 
+    @app.errorhandler(401)
+    def handle401(e):
+        return render_template('errors/401.html'), 401
 
-app = Flask(__name__)
-app.config.from_object('config.Configuration')
-Bootstrap(app)
-app.config['BOOTSTRAP_SERVE_LOCAL'] = True
 
-# use our own bootstrap theme
-app.extensions['bootstrap']['cdns']['bootstrap'] = StaticCDN()
+def add_routes(application):
+    # import views  # TODO convert to blueprint
+    # import views.stats  # TODO convert to blueprint
 
-db = SQLAlchemy(app)
+    from views.order import order_bp
+    from views.general import general_bp
+    from views.stats import stats_blueprint
+    from login import auth_bp
+    from zeus import oauth_bp
 
-toolbar = DebugToolbarExtension(app)
+    application.register_blueprint(general_bp, url_prefix='/')
+    application.register_blueprint(order_bp, url_prefix='/order')
+    application.register_blueprint(stats_blueprint, url_prefix='/stats')
+    application.register_blueprint(auth_bp, url_prefix='/')
+    application.register_blueprint(oauth_bp, url_prefix='/')
 
 
-if not app.debug:
-    timedFileHandler = TimedRotatingFileHandler(app.config['LOGFILE'], when='midnight', backupCount=100)
-    timedFileHandler.setLevel(logging.DEBUG)
+def add_template_filters(app):
+    @app.template_filter('countdown')
+    def countdown(value, only_positive=True, show_text=True):
+        delta = value - datetime.now()
+        if delta.total_seconds() < 0 and only_positive:
+            return "closed"
+        hours, remainder = divmod(delta.seconds, 3600)
+        minutes, seconds = divmod(remainder, 60)
+        time = '%02d:%02d:%02d' % (hours, minutes, seconds)
+        if show_text:
+            return 'closes in ' + time
+        return time
 
-    loglogger = logging.getLogger('werkzeug')
-    loglogger.setLevel(logging.DEBUG)
-    loglogger.addHandler(timedFileHandler)
-    app.logger.addHandler(timedFileHandler)
+    @app.template_filter('year')
+    def current_year(value):
+        return str(datetime.now().year)
 
-    airbrakelogger = logging.getLogger('airbrake')
+    @app.template_filter('euro')
+    def euro(value):
+        euro_string(value)
 
-    # Airbrake
-    airbrake = Airbrake(
-        project_id=app.config['AIRBRAKE_ID'],
-        api_key=app.config['AIRBRAKE_KEY']
-    )
-    # ugly hack to make this work for out errbit
-    airbrake._api_url = "http://errbit.awesomepeople.tv/api/v3/projects/{}/notices".format(airbrake.project_id)
 
-    airbrakelogger.addHandler(
-        AirbrakeHandler(airbrake=airbrake)
-    )
-    app.logger.addHandler(
-        AirbrakeHandler(airbrake=airbrake)
-    )
+# For usage when you directly call the script with python
+if __name__ == '__main__':
+    manager = create_app()
+    manager.run()
+    #app.run(threaded=True, host='0.0.0.0')
```

Comment:
Is this code still needed?

Reply:
The manager is for plugging in the migration commands into the application.
This is just the entrypoint.

Reply:
Oh, sorry, I meant the commented code

Reply:
I left out the threading part in the new config because the manager doesn't take that part.
I think it isn't needed but wasn't completely sure.

Reply:
see commit e5f16f7